packages/jira-adapter/src/filters/workflowV2/transition_parameters_filter.ts

```diff
@@ -52,14 +87,35 @@ const filter: FilterCreator = ({ config }) => ({
       .filter(element => element.elemID.typeName === JIRA_WORKFLOW_TYPE)
       .filter(isInstanceElement)
       .forEach(instance => {
-        instance.value.transitions.forEach((transition: Values) => {
-          transition.conditions?.conditions?.forEach((condition: Values) => {
-            convertParameters(condition?.parameters, CONDITION_LIST_FIELDS)
-          })
-          transition.validators?.forEach((validator:Values) => {
-            convertParameters(validator?.parameters, VALIDATOR_LIST_FIELDS)
-          })
-        })
+        convertTransitionParametersFieldsToList(instance.value.transitions)
+      })
+  },
+  preDeploy: async (changes: Change<ChangeDataType>[]) => {
+    if (!config.fetch.enableNewWorkflowAPI) {
+      return
+    }
+    changes
+      .filter(isInstanceChange)
+      .filter(isAdditionOrModificationChange)
+      .map(getChangeData)
+      .filter(instance => instance.elemID.typeName === JIRA_WORKFLOW_TYPE)
+      .forEach(instance => {
+        const workflow = instance.value.workflows[0]
+        convertTransitionParametersFieldsToString(workflow.transitions)
```

Comment:
I think we should be more careful with accessing properties in deploy 

```suggestion
if (Array.isArray(workflow?.transitions) {
     convertTransitionParametersFieldsToString(workflow.transitions)
}
```